### ISSUE TITLE
Cannot use shortcuts on BlSpace

### ISSUE THREAD

**Issue by labordep** [pharo-graphics/Bloc#355]
Try this:

```
space := BlSpace new show.
space addShortcut: (BlShortcutWithAction new
			              combination: [ (BlKeyCombination builder key: KeyboardKey F12) build ];
			              action: [ :a :b | self halt ]).
```

Press F12 in the space, the halt is never call.

**Comment by tinchodias** [comment]
Hi @labordep it is the first time I use this API! but I discovered that it was not matter of the halt... the handler is not executed. As it was not working , I started to gradually update it un til I wrote `root` as the receiver of `addShortcut:`. Check:

```
space := BlSpace new.
space root addShortcut:
	(BlShortcutWithAction new
		 combination: (BlSingleKeyCombination key: KeyboardKey F12);
		 action: [ :a :b | self inform: 'F12' ];
		 yourself).
space show.
```
I also removed the block closure from the combination, and the builder.

**Comment by tinchodias** [comment]
Okay, so the title of the issue is exact... my previous answer is only kind of a workaround, in the case it's blocking you... but in principple didn't understand why it didn't work on Space and it does work in the root. 

I have also a question: does it make sense to have BlSpace prividing this API, or should we remove it as we did with addChild? @plantec

**Comment by plantec** [comment]
the space event handler is actually used by the host to send BlSpaceEvent.
a space has a BlSpaceEventListener for those events.
so the space event related api is used. the issue is more a separation of concerns issue.
TBlEventTarget has api for both event handling and shortcuts handling

**Comment by labordep** [comment]
I agree with @plantec even if root not changed remains unchanged throughout the space's lifecycle. 
I don't know if this is normal behavior, I think yes. Taking this precaution is essential for creating a shortcut and preventing potential rework in case of root change.

**Comment by tinchodias** [comment]
Thanks Alain. May be BlSpace should be something like TBlEventTarget but without shortcuts API, like splitting the trait.

I see that when a key is pressed, the method `BlKeyboardProcessor>>#findTarget` determines which element will be "target" of the key event. For the space's shortcut handler... should the space be the target? I don't have the answer yet, but I smell they are polymorphic and it's fine that the root element is the target. 

In the other hand, you are right Pierre, this is part of the API:`BlSpace new root: BlElement new`

**Comment by tinchodias** [comment]
@plantec @labordep I continued the discussion in #702
